pkg/services/queryhistory/database.go

```diff
@@ -324,3 +325,74 @@ func (s QueryHistoryService) migrateQueries(ctx context.Context, user *models.Si
 
 	return len(queryHistories), len(starredQueries), nil
 }
+
+func (s QueryHistoryService) deleteStaleQueries(ctx context.Context, olderThan int64) (int, error) {
+	var rowsCount int64
+
+	err := s.SQLStore.WithDbSession(ctx, func(session *sqlstore.DBSession) error {
+		sql := `DELETE 
+			FROM query_history 
+			WHERE uid IN (
+				SELECT uid FROM (
+					SELECT uid FROM query_history
+					LEFT JOIN query_history_star
+					ON query_history_star.query_uid = query_history.uid
+					WHERE query_history_star.query_uid IS NULL
+				) AS q
+			) 	
+			AND query_history.created_at <= ?`
+
+		res, err := session.Exec(sql, strconv.FormatInt(olderThan, 10))
+		if err != nil {
+			return err
+		}
+
+		rowsCount, err = res.RowsAffected()
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return 0, err
+	}
+
+	return int(rowsCount), nil
+}
+
+func (s QueryHistoryService) unstarQueriesOfRemovedUsers(ctx context.Context) (int, error) {
+	var rowsCount int64
+
+	err := s.SQLStore.WithDbSession(ctx, func(session *sqlstore.DBSession) error {
+		sqlRemovedUsers := `DELETE 
```

Comment:
There is probably a better SQL than this, so if you have any suggestions on how to improve it, please let me know. I am basically trying to delete every row in `query_history_star` that had `org_id` and `user_id` that is no longer in `org_user` table. Another place where we could do this is in `user.go` and do it directly when we are deleting user. Hmmm, that might be a better place for this actually. Let me know what do you think!